### DISCUSSION TITLE
No Module/Attribute

### DISCUSSION THREAD

**Discussion by De-Been-Tech-Solutions** [mudler/LocalAI#888] [Q&A] [answered]
<pre>(gpt4all) [insertusernamehere@fedora python]$ python3 -c "import gpt4all; print(gpt4all.models.list_models())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'gpt4all' has no attribute 'models'
(gpt4all) [insertusernamehere@fedora python]$ python3 -c "import gpt4all_bindings.python; print(gpt4all.models.list_models())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'gpt4all_bindings'<pre>

**Comment by De-Been-Tech-Solutions** [comment]
There appears to be no modules nor attributes that get installed when using the following python command:

<pre>python3 install -e .<pre>

**Comment by mudler** [comment]
@NickDeBeenSAE did you had a look at the getting started guide? https://localai.io/basics/getting_started/

**Reply by De-Been-Tech-Solutions** [reply]
Yeah, its for an operating system that I don't have installed.

I am trying to build and run the program locally on my Fedora PC.

**Reply by De-Been-Tech-Solutions** [reply]
All commands though, were roughly the exact same.

**Reply by De-Been-Tech-Solutions** [reply]
<pre>docker-compose up -d --pull always
bash: docker-compose: command not found...
Install package 'docker-compose' to provide command 'docker-compose'? [N/y] y


 * Waiting in queue... 
 * Loading list of packages.... 
 * Downloading details about the software repositories.... 
 * Loading list of packages.... 
The following packages have to be installed:
 docker-compose-1.29.2-7.fc38.noarch	Multi-container orchestration for Docker
 python3-attrs-22.2.0-2.fc38.noarch	Python attributes without boilerplate
 python3-bcrypt-3.2.2-5.fc38.x86_64	Modern password hashing for your software and your servers
 python3-cached_property-1.5.2-9.fc38.noarch	A cached-property for decorating methods in Python classes.
 python3-certifi-2022.09.24-2.fc38.noarch	Python package for providing Mozilla's CA Bundle
 python3-docker-5.0.3-3.fc38.noarch	A Python library for the Docker Engine API
 python3-docker+ssh-5.0.3-3.fc38.noarch	Metapackage for python3-docker: ssh extras
 python3-docker-pycreds-0.4.0-15.fc38.noarch	Python bindings for the docker credentials store API
 python3-dockerpty-0.4.1-27.fc38.noarch	Python library to use the pseudo-tty of a docker container
 python3-docopt-0.6.2-25.fc38.noarch	Pythonic argument parser, that will make you smile
 python3-dotenv-0.21.1-2.fc38.noarch	Read key-value pairs from a .env file and set them as environment variables
 python3-dotenv+cli-0.21.1-2.fc38.noarch	Metapackage for python3-dotenv: cli extras
 python3-jsonschema-4.17.3-2.fc38.noarch	Implementation of JSON Schema validation for Python
 python3-paramiko-3.1.0-1.fc38.noarch	SSH2 protocol library for python
 python3-pyasn1-0.4.8-12.fc38.noarch	ASN.1 tools for Python 3
 python3-pynacl-1.5.0-2.fc38.x86_64	Python binding to the Networking and Cryptography (NaCl) library
 python3-pyrsistent-0.19.3-2.fc38.x86_64	Persistent/Functional/Immutable data structures
 python3-texttable-1.6.4-4.fc38.noarch	Python module for creating simple ASCII tables
 python3-websocket-client-1.3.3-3.fc38.noarch	WebSocket client for python
Proceed with changes? [N/y] y


 * Waiting in queue... 
 * Waiting for authentication... 
 * Waiting in queue... 
 * Downloading packages... 
 * Requesting data... 
 * Testing changes... 
 * Installing packages... 
Builds, (re)creates, starts, and attaches to containers for a service.

Unless they are already running, this command also starts any linked services.

The `docker-compose up` command aggregates the output of each container. When
the command exits, all containers are stopped. Running `docker-compose up -d`
starts the containers in the background and leaves them running.

If there are existing containers for a service, and the service's configuration
or image was changed after the container's creation, `docker-compose up` picks
up the changes by stopping and recreating the containers (preserving mounted
volumes). To prevent Compose from picking up changes, use the `--no-recreate`
flag.

If you want to force Compose to stop and recreate all containers, use the
`--force-recreate` flag.

Usage: up [options] [--scale SERVICE=NUM...] [--] [SERVICE...]

Options:
    -d, --detach               Detached mode: Run containers in the background,
                               print new container names. Incompatible with
                               --abort-on-container-exit.
    --no-color                 Produce monochrome output.
    --quiet-pull               Pull without printing progress information
    --no-deps                  Don't start linked services.
    --force-recreate           Recreate containers even if their configuration
                               and image haven't changed.
    --always-recreate-deps     Recreate dependent containers.
                               Incompatible with --no-recreate.
    --no-recreate              If containers already exist, don't recreate
                               them. Incompatible with --force-recreate and -V.
    --no-build                 Don't build an image, even if it's missing.
    --no-start                 Don't start the services after creating them.
    --build                    Build images before starting containers.
    --abort-on-container-exit  Stops all containers if any container was
                               stopped. Incompatible with -d.
    --attach-dependencies      Attach to dependent containers.
    -t, --timeout TIMEOUT      Use this timeout in seconds for container
                               shutdown when attached or when containers are
                               already running. (default: 10)
    -V, --renew-anon-volumes   Recreate anonymous volumes instead of retrieving
                               data from the previous containers.
    --remove-orphans           Remove containers for services not defined
                               in the Compose file.
    --exit-code-from SERVICE   Return the exit code of the selected service
                               container. Implies --abort-on-container-exit.
    --scale SERVICE=NUM        Scale SERVICE to NUM instances. Overrides the
                               `scale` setting in the Compose file if present.
    --no-log-prefix            Don't print prefix in logs.<pre>

**Reply by De-Been-Tech-Solutions** [reply]
This is after installing Docker.

**Reply by De-Been-Tech-Solutions** [reply]
Ok, I have moved on from GenossGPT to LocalAI, which doesn't build:

<pre>docker build -t localai
ERROR: "docker buildx build" requires exactly 1 argument.
See 'docker buildx build --help'.

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Start a build<pre>

**Reply by De-Been-Tech-Solutions** [reply]
Your page needs updated instructions because whats there already is not working at all.

**Reply by De-Been-Tech-Solutions** [reply]
And to further prove my point:

<pre>docker build -t localai .
[+] Building 0.2s (2/2) FINISHED                                 docker:default
 => [internal] load .dockerignore                                          0.1s
 => => transferring context: 2B                                            0.0s
 => [internal] load build definition from Dockerfile                       0.1s
 => => transferring dockerfile: 2B                                         0.0s
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3475166/Dockerfile: no such file or directory<pre>

**Reply by mudler** [reply]
looks like your docker/docker-compose version are very old - I'm not sure what OS you are running on, but I'd rather suggest you to follow the instruction here first: https://docs.docker.com/engine/install/ and configure and install docker based on your distribution.

**Reply by mudler** [reply]
For Fedora: https://docs.docker.com/engine/install/fedora/#uninstall-old-versions

**Reply by De-Been-Tech-Solutions** [reply]
Nope. My version of Docker is the latest as I installed it directly from its repository.

**Reply by De-Been-Tech-Solutions** [reply]
And plus, I built LocalAI directly but its giving me this when I gave it the ./local-ai command:

{"error":{"code":404,"message":"Cannot GET /","type":""}}

**Reply by mudler** [reply]
this looks correct. You need to install a model and use it with the OpenAI clients or you can build yours, the example here: https://localai.io/basics/getting_started/#example-use-gpt4all-j-model-with-docker-compose shows gpt4all and how to start to use it right after.

**Reply by De-Been-Tech-Solutions** [reply]
I am using gpt4all.

**Comment by mudler** [comment]
> docker-compose up -d --pull always
> bash: docker-compose: command not found...
> Install package 'docker-compose' to provide command 'docker-compose'? [N/y] y
> 
> 
>  * Waiting in queue... 
>  * Loading list of packages.... 
>  * Downloading details about the software repositories.... 
>  * Loading list of packages.... 
> The following packages have to be installed:
>  docker-compose-1.29.2-7.fc38.noarch	Multi-container orchestration for Docker


That's not latest for sure, see here:

```bash
$ docker-compose --version                                        
Docker Compose version 2.20.2
```

**Reply by De-Been-Tech-Solutions** [reply]
Fedora........ It can't install newer versions of docker.

**Reply by De-Been-Tech-Solutions** [reply]
<pre>docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.11.4
OpenSSL version: OpenSSL 3.0.9 30 May 2023<pre>

**Reply by De-Been-Tech-Solutions** [reply]
<pre>sudo dnf upgrade docker
AMDGPU 5.5.3 repository                         109  B/s | 162  B     00:01    
Errors during downloading metadata for repository 'amdgpu':
  - Status code: 404 for https://repo.radeon.com/amdgpu/5.5.3/rhel//main/x86_64/repodata/repomd.xml (IP: 13.82.220.49)
Error: Failed to download metadata for repo 'amdgpu': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
Ignoring repositories: amdgpu
Last metadata expiration check: 0:29:10 ago on Sat 12 Aug 2023 21:53:29.
Dependencies resolved.
Nothing to do.
Complete!<pre>

**Reply by mtutty** [reply, marked as answer]
@NickDeBeenSAE You might try https://get.docker.com/ instead of relying on Fedora's repo process.

**Reply by De-Been-Tech-Solutions** [reply]
Ok, time to check the version again.

**Reply by De-Been-Tech-Solutions** [reply]
Its version is 24.

Thats good.

I have built and tested the LocalAI by building it from source code however.

**Reply by De-Been-Tech-Solutions** [reply]
Here in this case:

https://localai.io/basics/build/
